st2common/st2common/models/api/action.py

```diff
@@ -273,3 +282,51 @@ def to_model(cls, execution):
         model.callback = getattr(execution, 'callback', dict())
         model.result = getattr(execution, 'result', None)
         return model
+
+
+class ActionExecutionStateAPI(BaseAPI):
+    """
+    System entity that represents state of an action in the system.
+    This is used only in tests for now.
+    """
+    model = ActionExecutionStateDB
+    schema = {
+        "title": "ActionExecutionState",
+        "description": "Execution state of an action.",
+        "type": "object",
+        "properties": {
+            "id": {
+                "description": "The unique identifier for the action execution state.",
+                "type": "string"
+            },
+            "execution_id": {
+                "type": "string",
+                "description": "ID of the action execution.",
+                "required": True
+            },
+            "query_context": {
+                "type": "object",
+                "description": "query context to be used by querier.",
+                "required": True
+            },
+            "query_module": {
+                "type": "string",
+                "description": "Name of the query module.",
+                "required": True
+            }
+        },
+        "additionalProperties": False
+    }
+
+    @classmethod
+    def from_model(cls, model):
```

Comment:
The super class impl of this method is pretty much the same. Can you use that instead?